src/Symfony/Component/Validator/Constraints/Collection.php

```diff
@@ -87,4 +87,31 @@ protected function getCompositeOption()
     {
         return 'fields';
     }
+
+    private static function isFieldsOption($options): bool
+    {
+        if (!\is_array($options)) {
+            return false;
+        }
+
+        foreach ($options as $optionOrField) {
+            if ($optionOrField instanceof Constraint) {
+                return true;
+            }
+
+            if (null === $optionOrField) {
+                return true;
```

Comment:
I think `null` could be both describing a field without constraints in the `fields` option or another option (`payload` is nullable for instance). So I would rather use `continue` here, to abstain on such value and check other ones instead of considering `null` as a positive decision (if nothing else decides, we will still end with a positive decision at the end of the loop)

Reply:
Yes, that makes sense, done.